unified-scheduler-pool/src/lib.rs

```diff
@@ -194,34 +199,422 @@ impl TaskHandler for DefaultTaskHandler {
 }
 
 impl<TH: TaskHandler> PooledScheduler<TH> {
     fn do_spawn(pool: Arc<SchedulerPool<Self, TH>>, initial_context: SchedulingContext) -> Self {
+        // we're hard-coding the number of handler thread to 1, meaning this impl is currently
+        // single-threaded still.
+        let handler_count = 1;
+
         Self::from_inner(
             PooledSchedulerInner::<Self, TH> {
-                id: pool.new_scheduler_id(),
-                pool,
+                thread_manager: ThreadManager::new(pool, handler_count),
             },
             initial_context,
         )
     }
 }
 
+impl<S: SpawnableScheduler<TH>, TH: TaskHandler> ThreadManager<S, TH> {
+    fn new(pool: Arc<SchedulerPool<S, TH>>, handler_count: usize) -> Self {
+        let (new_task_sender, new_task_receiver) = unbounded();
+        let (session_result_sender, session_result_receiver) = unbounded();
+
+        Self {
+            scheduler_id: pool.new_scheduler_id(),
+            pool,
+            handler_count,
+            new_task_sender,
+            new_task_receiver,
+            session_result_sender,
+            session_result_receiver,
+            session_result_with_timings: None,
+            scheduler_thread: None,
+            handler_threads: Vec::with_capacity(handler_count),
+            accumulator_thread: None,
+        }
+    }
+
+    fn execute_task_with_handler(
+        bank: &Arc<Bank>,
+        executed_task: &mut Box<ExecutedTask>,
+        handler_context: &HandlerContext,
+    ) {
+        debug!("handling task at {:?}", thread::current());
+        TH::handle(
+            &mut executed_task.result_with_timings.0,
+            &mut executed_task.result_with_timings.1,
+            bank,
+            executed_task.task.transaction(),
+            executed_task.task.task_index(),
+            handler_context,
+        );
+    }
+
+    fn propagate_context_to_handler_threads(
+        runnable_task_sender: &mut Sender<ChainedChannel<Task, SchedulingContext>>,
+        context: SchedulingContext,
+        handler_count: usize,
+    ) {
+        let (next_sessioned_task_sender, runnable_task_receiver) = unbounded();
+        for _ in 0..handler_count {
+            runnable_task_sender
+                .send(ChainedChannel::chain_to_new_channel(
+                    context.clone(),
+                    runnable_task_receiver.clone(),
+                ))
+                .unwrap();
+        }
+        *runnable_task_sender = next_sessioned_task_sender;
+    }
+
+    fn take_session_result_with_timings(&mut self) -> ResultWithTimings {
+        self.session_result_with_timings.take().unwrap()
+    }
+
+    fn put_session_result_with_timings(&mut self, result_with_timings: ResultWithTimings) {
+        assert_matches!(
+            self.session_result_with_timings
+                .replace(result_with_timings),
+            None
+        );
+    }
+
+    fn start_threads(&mut self, context: &SchedulingContext) {
```

Comment:
closures in this method is pretty full of unwraps on crossbeam channels. most of them will be addressed in later prs.

that's said, these unwraps should be acceptable even for production. That's because once scheduler is spawn, it never stops the backing threads.